src/st/nz1/e_cloak_knight.c

```diff
@@ -0,0 +1,441 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+#include "nz1.h"
+
+// n.b.! unsure of the correct number of values in this array,
+//       but it's more than 8 and less than 13.
+static u8 g_cloak_anim[] = {
+    4, 3, 4, 4, 4, 5, 4, 6, 0, 0, 0, 0,
+};
+
+#include "../step_towards.h"
+
+extern EInit g_EInitCloakKnight;
+
+void EntityCloakKnight(Entity* self) {
+    Entity* entity;
+    s32 primIndex;
+    Primitive* prim;
+    s32 posX;
+    s32 posY;
+    Pos* pos;
+    s32 distance;
+    s32 angle;
+    s32 scale;
+
+    if (self->flags & 0x100 && self->step != 6) {
+        SetStep(6);
+    }
+
+    switch (self->step) {
+    case 0:
+        InitializeEntity(g_EInitCloakKnight);
+        self->animCurFrame = 1;
+        self->hitboxOffY = 11;
+        primIndex = g_api.AllocPrimitives(PRIM_TILE, 1);
+        if (primIndex == -1) {
+            DestroyEntity(self);
+            return;
+        }
+        self->flags |= FLAG_HAS_PRIMS;
+        self->primIndex = primIndex;
+        prim = &g_PrimBuf[primIndex];
+        self->ext.prim = prim;
```

Comment:
I think this can be `self->ext.cloakKnight.prim` if you use 0x7C which is unused currently.

There's a few other uses down the file as well